system/java_certs.py

```diff
@@ -0,0 +1,167 @@
+
+# Remove certificate with given alias from a keystore
+java_certs: cert_url=google.com keystore_path=/usr/lib/jvm/jre7/lib/security/cacerts keystore_pass=changeit state=absent
+'''
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+import os
+import subprocess
+import sys
+import datetime
+import syslog
+
+def import_cert_url(module, url, port, keystore_path, keystore_pass, alias):
+    fetch_cmd = "keytool -printcert -rfc -sslserver %s:%s" % (url, port)
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Fetch SSL certificate from remote host.
+    (rc, fetch_out, fetch_err) = module.run_command(fetch_cmd, check_rc=True)
+
+    # Use remote certificate from remote host and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, data=fetch_out, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
+            return module.exit_json(changed=False, msg=import_out,
+                rc=0, cmd=import_cmd, stdout_lines=import_out)
+        else:
+            return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def import_cert_path(module, path, keystore_path, keystore_pass):
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -file '%s'" % (keystore_path, keystore_pass, path)
+
+    # Use local certificate from local path and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, check_rc=False)
+    if not rc:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        if 'already exists' in import_out:
+            return module.exit_json(changed=False, msg=import_out,
+                rc=0, cmd=import_cmd, stdout_lines=import_out)
+        else:
+            return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+
+def delete_cert(module, keystore_path, keystore_pass, alias):
+    del_cmd = "keytool -delete -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Delete SSL certificate from keystore
+    (rc, del_out, del_err) = module.run_command(del_cmd, check_rc=True)
+
+    return module.exit_json(changed=True, msg=del_out,
+        rc=rc, cmd=del_cmd, stdout_lines=del_out)
+
+def main():
+    argument_spec = dict()
+    argument_spec.update(dict(
+            cert_url = dict(required=False),
+            cert_path = dict(required=False),
+            cert_alias = dict(required=False),
+            cert_port = dict(required=False, default='443'),
+            keystore_path = dict(required=False, default='/usr/lib/jvm/java-7-oracle-amd64/jre/lib/security/cacerts'),
```

Comment:
this seems like an awfully specific default that won't work with many variations

I would either try to autodetect or make this a required option